output/cloud/expv2/flush_test.go

```diff
@@ -96,3 +160,79 @@ func (pm *pusherMock) push(_ string, _ *pbcloud.MetricSet) error {
 	pm.pushCalled++
 	return nil
 }
+
+func Test_tracesFlusher_Flush_ReturnsNoErrorWithWorkingInsightsClientAndNonCancelledContextAndNoData(t *testing.T) {
+	t.Parallel()
+
+	// Given
+	data := insights.RequestMetadatas{}
+	cli := &mockWorkingInsightsClient{}
+	col := &mockRequestMetadatasCollector{data: data}
+	flusher := newTracesFlusher(cli, col)
+
+	// When
+	err := flusher.flush(context.Background())
+
+	// Then
+	require.NoError(t, err)
+	require.False(t, cli.ingestRequestMetadatasBatchInvoked)
+	require.False(t, cli.dataSent)
+	require.Empty(t, cli.data)
+}
+
+func Test_tracesFlusher_Flush_ReturnsNoErrorWithWorkingInsightsClientAndNonCancelledContextAndData(t *testing.T) {
+	t.Parallel()
+
+	// Given
```

Comment:
Out of curiosity, why do you think this practice is helpful? 

If they work for you, I'm not against keeping them, but in my opinion, a unit test should try to achieve some level of auto-explanation and I think a test like the current already does.

The constructors are grouped, the logic method is only one, and the assert section is also grouped.

Reply:
At this point, it's muscle memory. It helps me when I revisit some bigger unit tests.